Kconfig.projbuild

```diff
@@ -82,6 +82,39 @@ choice ARDUINO_UDP_RUNNING_CORE
 
 endchoice
 
+choice ARDUINO_SERIAL_EVENT_TASK_RUNNING_CORE
```

Comment:
this option should depend on wether the chip has more than one core. Example implementation: https://github.com/espressif/esp-idf/blob/master/components/lwip/Kconfig#L675

Reply:
Hello @me-no-dev , I'm a bit confused about this suggestion. In the other config entries (about running cores) there's no dependency on this define, should this be added to every entry aswell?

Also, I did a search about that specific define, but I've found CONFIG_FREERTOS_UNICORE and not FREERTOS_UNICORE, is that a bug in the line you linked to me? 

I guess I could answer myself some of this questions, but I don't really know how to test this. I don't know what's the procedure to use this kconfig file. Could you point me to a guide?

Reply:
`CONFIG_` prefix is added to KConfig Options in its created header file automaticaly.
So Kconfig Menu option FREERTOS_UNICORE will be seen as a `#define CONFIG_FREERTOS_UNICORE` in the source code.

Reply:
I think we may need to review it in the whole Kconfig.projectbuild.

I also got the same doubt as @gonzabrusco ...
Shall we include this dependency for all not valid options when there is just one Core (S2/C3)?
Should the default option go to Core0 when there is just one Core?

Reply:
I think that if we use `xTaskCreateUniversal` it does not matter the default option. 

https://github.com/espressif/arduino-esp32/blob/0b3f1a9fa9bb5cc4df9169fe9130f324cbcba9db/cores/esp32/esp32-hal-misc.c#L148

Reply:
True that `xTaskCreateUniversal` will take care regardless of how many cores, but this is a safe-net inside the function itself. It still makes sense to hide options that are invalid for the current target in ESP-IDF. This is how it is done for all IDF components. Arduino is no different :)

> I think we may need to review it in the whole Kconfig.projectbuild.

Probably. It has not really been touched since ESP32, which has 2 cores always :D At least what we support for it.

Reply:
I'm on vacation now, so feel free to change what you want to move this forward.

Reply:
We are not in a hurry, so no worries :) Have fun at your vacation